client/ojo_client.go

```diff
@@ -0,0 +1,36 @@
+package client
+
+import (
+	"context"
+
+	"github.com/ojo-network/ojo/client/query"
+	"github.com/ojo-network/ojo/client/tx"
+	"github.com/rs/zerolog"
+)
+
+// OjoClient is a helper for initializing a keychain, a cosmos-sdk client context,
+// and sending transactions/queries to a specific Ojo node
+type OjoClient struct {
+	QueryClient *query.Client
+	TxClient    *tx.Client
+}
+
+func NewOjoClient(
+	chainID string,
+	tmrpcEndpoint string,
+	grpcEndpoint string,
+	accountName string,
+	accountMnemonic string,
+) (uc *OjoClient, err error) {
+	uc = &OjoClient{}
+	uc.QueryClient, err = query.NewQueryClient(grpcEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	uc.TxClient, err = tx.NewTxClient(chainID, tmrpcEndpoint, accountName, accountMnemonic)
+	return uc, err
+}
+
+func (oc *OjoClient) NewChainHeight(ctx context.Context, logger zerolog.Logger) (*ChainHeight, error) {
+	return NewChainHeight(ctx, oc.TxClient.ClientContext.Client, logger)
+}
```

Comment:
It would remove the need for this method but I think the little bit of extra code is worth keeping the separation of responsibilities. Otherwise the chain_height functions become specific to Ojo instead of just the cosmos sdk, and has to have knowledge of the structure of the OjoClient. I've been chatting with Robert about abstracting this client out to a non-node specific library so for right now it might be better to leave the chain_height file the way it is.

Reply:
works for me, seems like that way leads to less code in the long-term :^)